manual/manual.html

```diff
@@ -7894,6 +7925,48 @@ <h3 id="_words_identifiers_and_names"><a class="anchor" href="#_words_identifier
 End Verb 'look'.</code></pre>
 </div>
 </div>
+<div class="admonitionblock important">
+<table>
+<tr>
+<td class="icon">
+<i class="fa icon-important" title="Important"></i>
+</td>
+<td class="content">
+<div class="paragraph">
+<p>Whenever an identifier contains an Alan keyword, it&#8217;s mandatory to use a quoted identifier.</p>
+</div>
+</td>
+</tr>
+</table>
+</div>
+<div class="paragraph">
+<p>In the following example, the <code>'The Empty Room'</code> name needs to be stropped because it contains two Alan keywords (<code>The</code> and <code>Empty</code>), but there&#8217;s no need to stropp the <code>empty_room</code> identifier because the word <code>empty</code> doesn&#8217;t appear in isolated form (i.e. surrounded by spaces):</p>
+</div>
+<div class="listingblock">
+<div class="content">
+<pre class="highlightjs highlight"><code data-lang="alan" class="language-alan hljs">The empty_room IsA location   --&gt; No need to stropp 'empty_room'.
+  NAME 'The Empty Room'.      --&gt; Keywords 'THE' and 'EMPTY' stropped.
+End The.</code></pre>
+</div>
+</div>
+<div class="admonitionblock note">
+<table>
+<tr>
+<td class="icon">
+<i class="fa icon-note" title="Note"></i>
+</td>
+<td class="content">
+<div class="paragraph">
+<p>Stropping occurs only when the quoted identifier contains at least one Alan keyword.
```

Comment:
"occurs" -> "is only required"?

Reply:
What I was trying to say is that quoting an identifier doesn't always imply stropping. I wanted the reader not to confuse stropping with quoting identifiers. We have a case of stropping only if the quoted identifier is actually a reserved word — most likely a single word ID, but could be also a multi-word stropping (e.g. `'current location'`, etc).

Stropping is mandatory for using keywords as identifiers, but authors can freely use single quotes around identifiers any time (although no stropping occurs).

It was an attempt to clarify the context, but if it makes more confusing then it's not worth keeping it.

Reply:
I confirm: "occurs only when" as in "we have a case of stropping only when".

Reply:
I was not actually aware of the term "stropping" (and I was a compiler builder by trade, although quite a few years back now). So I'm not sure introducing this extra concept helps.

I think that what we should say in as few words and concepts as possible is "If you want an identifier that is either a keyword or contains characters that are not allowed in identifiers you can do that by putting that string inside quotes. and by the way, consider that the player might not be able to use those in the input."

Reply:
The term "stropping" seem _en vogue_ right now, with modern languages (among them Nim lang) promoting its use, so it earned a Wikipedia page for itself. Not sure who cointed it and when (hacker jargon is hard to trace).

> So I'm not sure introducing this extra concept helps.

I had mixed feeling myself and hesitated for a long time, but ultimately thought it might help actually. On the one hand, I think that is a good idea to introduce some technical terms here and there, possibly in admonitions tips and notes, to help the non-programmer author embrace some programming concepts too, for he/she might eventually like to dig more into them.

On the other hand, terms like "escaping" and "stropping" can simplify communication of the Yahoo group. Just think of the `"$+1"` notation, for which we have no term to refer to right now. Some of these "dollar symbols" act as escape sequences, others are more like interpolations (as in `printf` notation). Referring to them as "dollars", "dollar symbols", "parameters special symbols" tend to make questions and answers harder to handle.

So, ultimately, some technical jargon could help pin-point specific uses of Alan syntax constructs — in this case, stropping is a specific use case of quoted identifiers, and even though knowing how "quoted IDs" work is what really matters to the reader/author, the term "stropping" is a reminder of the need to quote keywords.

In a way, giving names to things is the key to all knowledge (or so claims the Old Testament).

> I think that what we should say in as few words and concepts as possible is "If you want an identifier that is either a keyword or contains characters that are not allowed in identifiers you can do that by putting that string inside quotes. and by the way, consider that the player might not be able to use those in the input."

The text does say so, but I thought that "stropping" was a good section title (instead of "Using keywords inside identifiers"), and would give more visibility to the topic.

But I agree with you, if introducing technical terms is going to be a burden on the reader we shouldn't do it. The ideal thing would be polling the community on this — some might feel that techinical terms are a barrier, but others might feel that they mitigate the gap between programmer Alan authors and non-programmer authors (a frequent issue in the IF community).

We could remove Stropping from the title, and keep the mention of the term (which is also in the Glossary, which doesn't harm the contents)-

<!-- EOF -->

Reply:
I totally agree with the need for terms, I was rather questioning the need for the distinction in this context remembering the audience.

I think your last suggestion seems like good middle ground. Removing the word from the title (and instead use a more obvious one), refraining from putting stress on the term, but use it and have an entry in the glossary.